cpp/src/arrow/record_batch.cc

```diff
@@ -276,7 +284,8 @@ struct ConvertColumnsToTensorVisitor {
 };
 
 template <typename DataType>
-inline void ConvertColumnsToTensor(const RecordBatch& batch, uint8_t* out) {
+inline void ConvertColumnsToTensor(const RecordBatch& batch, uint8_t* out,
+                                   bool null_to_nan) {
```

Comment:
The `null_to_nan` is not used here?

Reply:
Oh no, of course. No need for it anymore due to the visitor pattern. Will remove!